src/app/TargetSelect/TargetSelect.tsx

```diff
@@ -79,12 +80,16 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props)
   React.useEffect(() => {
     try {
       const previouslyCachedTarget = getCachedTargetSelection();
-      context.target.setTarget(previouslyCachedTarget);
+      const targetStillExists = targets.some(target => _.isEqual(previouslyCachedTarget, target));
+
+      if(targetStillExists) {
+        context.target.setTarget(previouslyCachedTarget);
+      }
```

Comment:
Would it make sense to add an `else` here that removes the entry from the cache?

Reply:
It looks like the `useEffect` hook gets triggered twice while the `targets` list is still empty.  Removing entries from the cache might remove a correct entry before `targets` is populated. I'll see if I can restructure the code to make it easier to read.
```
console.log("use effect triggered with target list: ", targets)
      if(targetStillExists) {
        context.target.setTarget(previouslyCachedTarget);
      }
```
![image](https://user-images.githubusercontent.com/84587295/136234386-6bba846b-3d77-45c1-a197-4b4298d10cce.png)

Reply:
I moved the `setTarget` logic inside a callback function that only runs whenever the observable `Targets[]` changes, then added the `else` statement to remove the invalid cache entry